snapcraft/_store.py

```diff
@@ -266,6 +266,28 @@ def login_decorator(self, *args, **kwargs):
     return login_decorator
 
 
+def _register_wrapper(method):
+    def register_decorator(self, *args, snap_name: str, **kwargs):
+        try:
+            return method(self, *args, snap_name=snap_name, **kwargs)
+        except storeapi.errors.StorePushError as push_error:
+            if "resource-not-found" not in push_error.error_list:
+                raise
+            echo.wrapped(
+                "You are required to register this snap before continuing. "
+                "Refer to 'snapcraft help register' for more options."
```

Comment:
Maybe only print `Refer to 'snapcraft help register' for more options.` if they say no to registration?

Reply:
Let's see how this plays out, I do not want to give the impression that the option shown is the only way to register a snap and coerce people into registering something that they may have wanted to be private.